esp32_serial_node:
  ros__parameters:
    # Hardware Settings
    motor_max_rpm: 330          # rpm of motor when fully operated
    wheel_base: 0.465           # (m) distance between wheels
    wheel_radius: 0.02          # (m) radius of wheels
    encoder_cpr: 780.0          # count per revolution, (13 PPR * 30 gear ratio * 4 encoders tick per revolution / 2 for only checking phase A = 700)

    # Slip Ratio
    param_n: 0.5                # parameter to compensate speed difference between wheels

    # Serial Communication
    serial_port: "/dev/esp32"
    serial_baudrate: 115200
    serial_timeout: 1.0
    
    # Timer Settings
    serial_timer: 0.01           # refresh rate for serial communication with ESP32
    odom_update_interval: 0.05    # refresh rate for publish odom topic to ROS

    # Frame IDs
    odom_frame_id: "odom"
    base_frame_id: "base_footprint"
    
    # Topic Names
    cmd_vel_topic: "/cmd_vel"
    odom_topic: "/odom"
